### DISCUSSION TITLE
Circuit Information

### DISCUSSION THREAD

**Discussion by concipaulo** [theOehrly/Fast-F1#129] [General]
Hello, 

First, thanks for the amazing module. Now, I would like to know if there's anyway to retrieve circuit information, like corner distances, Speed Trap and DRS Detection location in relation to distance traveled from lap start. 

Thanks.

**Comment by theOehrly** [comment]
I don't think you will find that information easily. 

There's a circuit map published for each race on the FIA website. It contains some of that information, but no distance values or anything like that. So you'd need to approximate the distance manually somehow.

https://www.fia.com/documents/championships/fia-formula-one-world-championship-14/season/season-2022-2005

**Reply by concipaulo** [reply]
Yeah that's the exact plot I'm trying to recreate. I look over the most known f1 sites, but no luck.

I actually was able to find those distances manually, plotting an iterative graph with x and y coordinates from telemetry data. The thing is, those coordinates only exists from that lap, and you could not use for any other. So I would have to do that for every lap if I want to compare with any other.

You do plot similar graphs [here](https://theoehrly.github.io/Fast-F1/examples_gallery/plot_gear_shifts_on_track.html#sphx-glr-examples-gallery-plot-gear-shifts-on-track-py) which is very cool. I'm thinking that if I could find the steering angle for some amount of distance and if this angle is positive or negative I possibly could identify if the corner is right or left, no change for some amount of distance means a straight, this is just an idea tho. 

The thing is, I quite don't understand these two lines of code
```
points = np.array([x, y]).T.reshape(-1, 1, 2)
segments = np.concatenate([points[:-1], points[1:]], axis=1)
```
why are you transposing and reshaping the coordinates and these segments are lines perpendicular to the circuit? 

Anyways, thanks for your time and your awesome work. Good race tomorrow mate :)

**Reply by theOehrly** [reply]
I don't really have a solution for your problem with the distances, sorry.

Yes, that corner detection might very well work. I've thought about it, but never took the time to try it. 
Those two lines of code are just, well, there to create the line segments. You can probably write it different (and shorter, actually). Maybe I should improve it, for readability’s sake.
The LineCollection is documented here: https://matplotlib.org/stable/api/collections_api.html#matplotlib.collections.LineCollection
The code creates one line segment between every point and the point after it.  The line segements are not in fact perpendicular to the circuit. They're just wider than they are long.

You could just create a scatter plot with coloured markers. But then you don't get the single continuous line, so it does not look as good.
